### PR TITLE
6 - product accordion

### PR DESCRIPTION
**Pull request by ogavashi**

### CODE REVIEW

**Comment by ogavashi** [comment]
image
Na

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import AddRoundedIcon from '@mui/icons-material/AddRounded';
+import SwapHorizIcon from '@mui/icons-material/SwapHoriz';
+import CloseRoundedIcon from '@mui/icons-material/CloseRounded';
+import useStyles from './styles';
+import globalUseStyles from '../../../../../common/globalUseStyles';
+
+export type BuilderProductSpecs = {
+  name: string;
+  value: string;
+};
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx]: BuilderProductSpec

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+type ProductProps = {
+  id: string;
+  title: string;
+  mainImage: string;
+  price: number;
+  specs: BuilderProductSpecs[];
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx]: you should move this to separate type - BuildProduct

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      <Box className={styles.leftWrapper}>
+        <img className={styles.image} src={mainImage} alt={title} />
+        <Box>
+          <Typography variant="h5">{title}</Typography>
+          <Box>
+            {specs.map((spec) => (
+              <Box key={spec.name} className={styles.specsWrapper}>
+                <Typography
+                  variant="h6"
+                  paddingRight={1}
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx]: marginRight

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        />
+      </IconButton>
+    ) : (
+      <IconButton onClick={() => handleSelect(id)}>
+        <SwapHorizIcon
+          className={globalStyles.greenBuilderIcon}
+          fontSize="large"
+        />
+      </IconButton>
+    );
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx]: why function? variable can be JSX

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/styles.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      margin: theme.spacing(5),
+    },
+    specsWrapper: {
+      display: 'flex',
+      flexDirection: 'row',
+      alignItems: 'center',
+    },
+    price: {
+      fontSize: '40px',
+    },
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/styles.ts]: this should be described in theme variants

File: src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+export type BuilderProductSpecs = {
+  name: string;
+  value: string;
+};
+
+type ProductProps = {
+  id: string;
+  title: string;
+  mainImage: string;
+  price: number;
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/BuilderProduct/index.tsx]: we dont have price at this time

File: src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+];
+
+const CPUBuilder: React.FC = () => {
+  const [selectedId, setSelectedId] = useState<string>('');
+  const [expand, setExpand] = useState<boolean>(false);
+
+  const onClickAdd = (productId: string) => {
+    setExpand(false);
+    setSelectedId(productId);
+  };
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx]: handleAddProduct

File: src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+  const toggleAccordion = () => {
+    setExpand((prev) => !prev);
+  };
+
+  const products = parts.map((part) => (
+    <BuilderProduct
+      {...part}
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/CPUBuilder/index.tsx]: it is better not to destructurize - easier to maintain in the future

File: src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    specs: [
+      { name: 'Socket', value: 'AM4' },
+      { name: 'Series', value: 'AMD Ryzen' },
+    ],
+  },
+];
+
+type ProductAccordionProps = {
+  icon: React.FC;
+  category: string;
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx:51]: category shold have pipe type

Comment (VHarastei) [reply src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx:51]: CPUs for example (check name of collections in types)

File: src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+            src={selectedProduct?.mainImage}
+            alt={selectedProduct?.title}
+          />
+        ) : (
+          <Icon />
+        )}
+        <Typography variant="h5" className={styles.title}>
+          {selectedId
+            ? ` ${selectedProduct?.title}`
+            : `${category} is not selected`}
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx]: selectedId? selectedProduct.name : `${category} is not selected`

File: src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+import AddRoundedIcon from '@mui/icons-material/AddRounded';
+import SwapHorizIcon from '@mui/icons-material/SwapHoriz';
+import RemoveRoundedIcon from '@mui/icons-material/RemoveRounded';
+import useStyles from './styles';
+import globalUseStyles from '../../../../../common/globalUseStyles';
+
+const parts = [
+  {
+    id: '1',
+    title: 'Ryzen 5 3600',
```
Comment (VHarastei) [inline src/components/Unknown/BuildScreen/Builder/ProductAccordion/index.tsx]: name

**Review by VHarastei — APPROVED** [review]

**Comment by VHarastei** [comment]
lgtm

**Pull request merged**